lib/runner/variant-reassignment.js

```diff
@@ -83,10 +83,13 @@ export default class VariantReassignment {
     // check if product types are the same for productDraft and CTP product to update
     const ctpProductTypeId = ctpProductToUpdate.productType.id
     const draftProductType = productDraft.productType.id
-    if (draftProductType !== ctpProductTypeId)
+    if (draftProductType !== ctpProductTypeId) {
       ctpProductToUpdate = await this._changeProductType(
         transaction, ctpProductToUpdate, draftProductType
       )
+      // find and replace ctpProductToUpdate in matchingProducts array with updated version
+      matchingProducts = this._replaceProductInProductArray(ctpProductToUpdate, matchingProducts)
```

Comment:
Do we need `ctpProductToUpdate` to be in `matchingProducts`? Sometimes it can be missing - eg if the process fails when changing product type - we delete `ctpProductToUpdate` but it won't be created successfully, the `ctpProductToUpdate` will have to be created again when resolving unfinished transaction and it won't be loaded in matching products

Reply:
Also when recreating `ctpProductToUpdate` [here](https://github.com/commercetools/commercetools-node-variant-reassignment/pull/45/files#diff-eda63119a30637151d12307cefd81cf0R80) we should save the result to `ctpProductToUpdate`?

Reply:
I think `ctpProductToUpdate` is also a matching product. If it's missing, it should not make any problems.

Reply:
It's not necessary to save result to `ctpProductToUpdate`, because the only thing that got updated hre is productId and version, which I don't use to compare anywhere.